plugins/Actions/VisitorDetails.php

```diff
@@ -0,0 +1,223 @@
+            case Piwik::LABEL_ID_GOAL_IS_ECOMMERCE_CART:
+                $action['icon'] = 'plugins/Morpheus/images/' . $action['type'] . '.png';
+                break;
+            case Action::TYPE_DOWNLOAD:
+                $action['type'] = 'download';
+                $action['icon'] = 'plugins/Morpheus/images/download.png';
+                break;
+            case Action::TYPE_OUTLINK:
+                $action['type'] = 'outlink';
+                $action['icon'] = 'plugins/Morpheus/images/link.png';
+                break;
+            case Action::TYPE_SITE_SEARCH:
+                $action['type'] = 'search';
+                $action['icon'] = 'plugins/Morpheus/images/search_ico.png';
+                break;
+            case Action::TYPE_EVENT:
+                $action['type'] = 'event';
+                $action['icon'] = 'plugins/Morpheus/images/event.png';
+                break;
+            case Action::TYPE_PAGE_URL:
+            case '':
+                $action['type'] = 'action';
+                $action['icon'] = null;
+                break;
+        }
+
+        // Convert datetimes to the site timezone
+        $dateTimeVisit = Date::factory($action['serverTimePretty'], Site::getTimezoneFor($visitorDetails['idSite']));
+        $action['serverTimePretty'] = $dateTimeVisit->getLocalized(Date::DATETIME_FORMAT_SHORT);
+        $action['timestamp'] = $dateTimeVisit->getTimestamp();
+
+        unset($action['idlink_va']);
+    }
+
+    /**
+     * @param $idVisit
+     * @return array
+     * @throws \Exception
+     */
+    protected function queryActionsForVisit($idVisit, $idSite)
+    {
+        $actionsLimit = (int)Config::getInstance()->General['visitor_log_maximum_actions_per_visit'];
+        $customFields = array();
+
+        Piwik::postEvent('Actions.getCustomActionDimensionFields', array(&$customFields, $idSite));
+
+        $customFields = array_filter($customFields);
+        array_unshift($customFields, ''); // add empty element at first
+        $sqlCustomVariables = implode(', ', $customFields);
+
+        // The second join is a LEFT join to allow returning records that don't have a matching page title
+        // eg. Downloads, Outlinks. For these, idaction_name is set to 0
+        $sql = "
+				SELECT
+					COALESCE(log_action_event_category.type, log_action.type, log_action_title.type) AS type,
+					log_action.name AS url,
+					log_action.url_prefix,
+					log_action_title.name AS pageTitle,
+					log_action.idaction AS pageIdAction,
+					log_link_visit_action.idlink_va,
+					log_link_visit_action.server_time as serverTimePretty,
+					log_link_visit_action.time_spent_ref_action as timeSpentRef,
+					log_link_visit_action.idlink_va AS pageId,
+					log_link_visit_action.custom_float,
+					log_link_visit_action.interaction_position
+					" . $sqlCustomVariables . ",
+					log_action_event_category.name AS eventCategory,
+					log_action_event_action.name as eventAction
```

Comment:
FYI: Not needed for now but eventually we also need to fetch `log_link_visit_action.idpageview` as other plugins will likely hook into individual actions based on the idpageview. I will add this likely once needed